src/test/conditions/Der_test.cpp

```diff
@@ -0,0 +1,1054 @@
+                 std::make_pair(-1, v{2, 1, -1}),
+                 std::make_pair(-2, v{2, 1, -2}),
+                 std::make_pair(std::int32_t(0xffffff00), v{2, 2, -1, 0}),
+                 std::make_pair(
+                     std::uint32_t(0xfffffffe), v{2, 5, 0, -1, -1, -1, -2}),
+                 std::make_pair(210, v{2, 2, 0, -46}),
+                 std::make_pair(0x101, v{2, 2, 1, 1}),
+                 std::make_pair(0x1000, v{2, 2, 16, 0}),
+                 std::make_pair(0x10001, v{2, 3, 1, 0, 1}),
+                 std::make_pair(0x100000, v{2, 3, 16, 0, 0}),
+                 std::make_pair(0x1001001, v{2, 4, 1, 0, 16, 1}),
+                 std::make_pair(0x1001001, v{2, 4, 1, 0, 16, 1})}};
+
+            cryptoconditions::der::TraitsCache dummy;
+            for (auto i = cases.begin(), e = cases.end(); i != e; ++i)
+            {
+                using Traits =
+                    cryptoconditions::der::DerCoderTraits<std::int64_t>;
+                this->BEAST_EXPECT(
+                    Traits::compare(i->first, i->first, dummy) == 0);
+                for (auto j = i + 1; j != e; ++j)
+                {
+                    this->BEAST_EXPECT(
+                        boost::math::sign(
+                            Traits::compare(i->first, j->first, dummy)) ==
+                        boost::math::sign(vecCmp(i->second, j->second)));
+                }
+            }
+        }
+    }
+
+    void
+    testString()
+    {
+        testcase("octet string");
+        auto makeTestCase = [this](
+            size_t n,
+            std::vector<char> const& expectedHeader,
+            char fillChar) -> std::pair<std::string, std::vector<char>> {
+            std::string s;
+            s.resize(n);
+            std::fill_n(s.begin(), n, fillChar);
+            std::vector<char> expected(expectedHeader);
+            auto const headerEnd = expected.size();
+            expected.resize(headerEnd + n);
+            std::fill_n(&expected[headerEnd], n, fillChar);
+            return std::make_pair(s, expected);
+        };
+
+        std::vector<std::pair<std::string, std::vector<char>>> cases = {{
+            makeTestCase(0, {4, 0}, 'a'),
+            makeTestCase(0, {4, 0}, 'z'),
+            makeTestCase(1, {4, 1}, 'a'),
+            makeTestCase(1, {4, 1}, 'z'),
+            makeTestCase(127, {4, 127}, 'a'),
+            makeTestCase(127, {4, 127}, 'z'),
+            makeTestCase(128, {4, -127, -128}, 'a'),
+            makeTestCase(128, {4, -127, -128}, 'z'),
+            makeTestCase(66000, {4, -125, 1, 1, -48}, 'a'),
+            makeTestCase(66000, {4, -125, 1, 1, -48}, 'z'),
+        }};
+
+        cryptoconditions::der::TraitsCache dummy;
+        for (auto i = cases.begin(), e = cases.end(); i != e; ++i)
+        {
+            test(i->first, i->second);
+            using Traits = cryptoconditions::der::DerCoderTraits<std::string>;
+            this->BEAST_EXPECT(Traits::compare(i->first, i->first, dummy) == 0);
```

Comment:
any particular reason for `this->BEAST_EXPECT(...` vs. just `BEAST_EXPECT(...`? It doesn't bother me either way, just wondering.